src/Simulation/CSharpGeneration/SimulationCode.fs

```diff
@@ -591,6 +592,9 @@ module SimulationCode =
 // TODO: diagnostics.
             | _ -> failwith ""
 
+        and buildSizedArray value size =
+            ident "QArray" <.> (ident "Repeat", [ buildExpression value; buildExpression size ])
```

Comment:
I think this may cause issues; if the given item to initialize the array with is an array itself, then you would need to formally copy it n times (meaning the reference count needs to be updated to reflect its use in all places).

Reply:
You're right that I needed to increment the ref count, but I think it's sound if the entire array owns one reference to the value. Since array indices are immutable, I don't think there is a way to mutate `xs[x][y]` in place. For example,

```qsharp
mutable items = [[1], size = 2];
set items w/= 0 <- items[0] w/ 0 <- 2;
```

will make a copy of `items[0]` before assigning it in-place to index 0 of `items`. Can you think of an example where this isn't true?

Reply:
Discussed offline: overwriting an array index can decrement the reference count, causing references in other indices to have reference count 0, so the count needs to be incremented initially for each index.